test/Solvers/ErrorMethods/full_residual.jl

```diff
@@ -0,0 +1,82 @@
+module residual_error 
+using Test, RLinearAlgebra, Random
+import LinearAlgebra: mul!, norm
+using ..FieldTest
+using ..ApproxTol
+Random.seed!(1232)
+
+mutable struct TestSolver <: Solver
+
+end
+
+mutable struct TestSolverRecipe <: SolverRecipe
+    solution_vec::AbstractVector
+end
+
+@testset "Full Residual" begin
+    @testset "Full Residual: SolverError" begin
+        # Verify Supertype
+        @test supertype(FullResidual) == SolverError 
+
+        # Verify fieldnames and types
+        @test fieldnames(FullResidual) == ()
+        @test fieldtypes(FullResidual) == ()
+        # Verify the internal constructor
+
+    end
+
+    @testset "Full Residual: SolverErrorRecipe" begin
+        # Verify Supertype
+        @test supertype(FullResidualRecipe) ==  SolverErrorRecipe
+
+        # Verify fieldnames and types
+        @test fieldnames(FullResidualRecipe) == (:residual,)
+        @test fieldtypes(FullResidualRecipe) == (AbstractVector,)
+    end
+
+    @testset "Residual: Complete error" begin
+        let n_rows = 4,
+            n_cols = 3
+            for type in [Float32, Float64, ComplexF32, ComplexF64] 
+                A = rand(type, n_rows, n_cols)
+                b = rand(type, n_rows)
+                x = rand(type, n_cols)
+                solver_rec = TestSolverRecipe(x)
```

Comment:
`A`, `b`, `x`, `solver_rec` should be arguments to the `let` block (probably `TestSolver()` and `FullResidual()` as well). Put the `for` loop outside of the `let`. Each set of tests on `error_rec` is then independent (though the for loop scopes it out).